packages/runtime/container-runtime/package.json

```diff
@@ -77,6 +77,8 @@
     "@fluidframework/runtime-utils": ">=2.0.0-internal.1.1.0 <2.0.0-internal.2.0.0",
     "@fluidframework/telemetry-utils": ">=2.0.0-internal.1.1.0 <2.0.0-internal.2.0.0",
     "double-ended-queue": "^2.1.0-0",
+    "lz4js": "^0.2.0",
+    "pako": "^2.0.4",
```

Comment:
I will appreciate help / ideas here.

Reply:
To deal with optional dependencies like this I see two approaches:
1. [Dynamic imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import)
2. Configurable compression through dependency injection

I'm personally partial to the dynamic imports approach, but I think an argument could be made for dependency injection. With dynamic imports, webpack will split the dynamically imported packages to separate chunks and will only download them when requested. This could lead to some sort of flow where we could begin loading the compression library after we first see a compressed op, or possibly when we see that compression is configured on the runtime.